docker/recorder/src/main/java/org/arquillian/cube/docker/impl/client/reporter/TakeDockerEnvironment.java

```diff
@@ -186,6 +201,59 @@ private GroupEntry createDockerCompositionSchema(CubeDockerConfiguration cubeDoc
         return containersComposition;
     }
 
+    private GroupEntry createNetworkTopologyGraph(CubeDockerConfiguration cubeDockerConfiguration, ReporterConfiguration reporterConfiguration, DockerClientExecutor executor) {
+
+        final GroupEntry netTopology = new GroupEntry("Network Topology Graph");
+        mxGraph graph = new mxGraph();
+        Object parent = graph.getDefaultParent();
+
+        graph.setAutoSizeCells(true);
+        graph.getModel().beginUpdate();
+        try {
+            DockerCompositions dockerCompositions = cubeDockerConfiguration.getDockerContainersContent();
+            final Map<String, CubeContainer> containers = dockerCompositions.getContainers();
+            final Map<String, Object> insertedVertex = new HashMap<>();
+                for (Map.Entry<String, CubeContainer> container: containers.entrySet()) {
+                    final String containerId = container.getKey();
+                    Object containerName = graph.insertVertex(parent, null , containerId, 0, 0, 80, 30);
+                    final CubeContainer cubeContainer = container.getValue();
+                    Set<String> nwList = new HashSet<>();
+                    if (cubeContainer.getNetworkMode() != null) {
+                        nwList.add(cubeContainer.getNetworkMode());
+                    } else {
+                        InspectContainerResponse inspect = executor.inspectContainer(containerId);
```

Comment:
In case of not setting `networkMode` in `docker-compose`, would docker host assign automatically a networkMode? I ask it because you are querying docker host about this.